chore(na): add some extra notes to the first time contributor guide (#1853)
medic/cht-docs: content/en/community/contributing/first-time-contributors.md

    git switch -c 1234-add-alonzo-church
    ```

>     _NOTE: Ensure your branch name contains the ticket number, as GitHub will automatically link it (and the eventual PR) to the ticket. This helps community members see that the issue is already being addressed._

Reviewer: Is this true for just the banch name of `1234-add-alonzo-church`?  I've only known GH to link to the ticket if:
* you have the ticket in the commit comment: `add alonzo church per #1234`
* mention the ticket in the PR body `Closes #1234`
* mention it in a comment in the PR

Lemme know if I'm wrong though!

however, it's still a valid point, so I think we can say:

```suggestion
    _NOTE: Ensure your branch name contains the ticket number.  This helps the community easily tell which ticket the branch applies to._
```

Author: I tried this out. I have a few findings. All the points that you mentioned are also valid, though.
1. Using the `create a branch` feature on GitHub's issue page, the branch is automatically linked to the issue.
![image](https://github.com/user-attachments/assets/df0855b8-e105-4a06-bee4-256b58f59598)

![image](https://github.com/user-attachments/assets/a4e2ccf2-3bad-4657-a1ff-b91df1fd934d)

2. Branches can be manually linked to an issue. _I was under the impression that having the ticket number and title would do that._
    
![image](https://github.com/user-attachments/assets/2aff7627-e58b-4c0a-8972-4df9d08d30ef)

I'll rephrase this note accordingly.

Author: On second thought, @mrjones-plip, I've decided to remove this note. The contribution [instructions](https://github.com/medic/cht-docs/blob/main/content/en/community/contributing/first-time-contributors.md#your-first-issue) here, it's specified that one should request to be assigned to the issue initially. I think that should indicate the state of the task.

Reviewer: ohhh!  That totally makes sense, thanks for the explanation with screen shots and all.
